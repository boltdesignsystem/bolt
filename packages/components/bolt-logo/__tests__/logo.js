import { render, stopServer, html } from '../../../testing/testing-helpers';

const timeout = 90000;

describe('logo', () => {
  let page;

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  afterAll(async function() {
    await stopServer();
    await page.close();
  });

  test('Basic usage', async () => {
    const results = await render('@bolt-components-logo/logo.twig', {
      src: '/fixtures/logo-paypal.svg',
      lazyload: false,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Logo with invert set to "true" renders properly', async () => {
    const results = await render('@bolt-components-logo/logo.twig', {
      src: '/fixtures/logo-paypal.svg',
      lazyload: false,
      invert: true,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-logo> renders', async function() {
    const renderedLogoHTML = await page.evaluate(() => {
      const logo = document.createElement('bolt-logo');

      // TODO: Use the actual logo web component once converted
      logo.innerHTML = `<bolt-image src="/fixtures/logo-paypal.svg" alt="Paypal Logo" ratio="124/33"></bolt-image>`;

      document.body.appendChild(logo);

      return logo.outerHTML;
    });

    const renderedHTML = await html(renderedLogoHTML);
    await page.waitFor(500);

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Inverted <bolt-logo> renders as inverted', async function() {
    const renderedLogoHTML = await page.evaluate(() => {
      const logo = document.createElement('bolt-logo');
      const wrapper = document.createElement('div');

      // TODO: Use the actual logo web component once converted
      logo.innerHTML = `<bolt-image src="/fixtures/logo-paypal.svg" alt="Paypal Logo" ratio="124/33"></bolt-image>`;
      logo.classList.add('c-bolt-logo--invert');

      wrapper.classList.add('t-bolt-dark');
      wrapper.appendChild(logo);

      document.body.appendChild(wrapper);

      return logo.outerHTML;
    });

    await page.evaluate(async () => {
      const selectors = Array.from(document.querySelectorAll('bolt-logo'));
      await Promise.all(
        selectors.map(logo => {
          const logoImage = logo.querySelector('bolt-image');
          if (logoImage._wasInitiallyRendered === true) return;
          return new Promise((resolve, reject) => {
            logoImage.addEventListener('ready', resolve);
            logoImage.addEventListener('error', reject);
          });
        }),
      );
    });

    const renderedHTML = await html(renderedLogoHTML);
    await page.waitFor(500);

    expect(renderedHTML).toMatchSnapshot();
  });
});
