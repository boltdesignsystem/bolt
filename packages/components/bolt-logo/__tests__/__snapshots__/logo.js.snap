// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logo Basic usage 1`] = `
<bolt-logo>
  <bolt-image no-lazy
              ratio="124/33"
              placeholder-color="hsl(233, 33%, 97%)"
              placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
              src="/fixtures/logo-paypal.svg"
              srcset="/fixtures/logo-paypal.svg"
  >
    <bolt-ratio ratio="124/33">
      <ssr-keep for="bolt-ratio"
                class="c-bolt-ratio"
                style="padding-bottom: calc((33 / 124) * 100%); --aspect-ratio: 3.7575757575758;"
      >
        <img class="c-bolt-image__image"
             alt
             srcset="/fixtures/logo-paypal.svg"
             sizes="auto"
             src="/fixtures/logo-paypal.svg"
        >
      </ssr-keep>
    </bolt-ratio>
  </bolt-image>
</bolt-logo>
`;

exports[`logo Default <bolt-logo> renders 1`] = `
<bolt-logo
  style=""
>
  <bolt-image
    alt="Paypal Logo"
    ratio="124/33"
    src="/fixtures/logo-paypal.svg"
    style=""
  />
</bolt-logo>
`;

exports[`logo Inverted <bolt-logo> renders as inverted 1`] = `
<bolt-logo
  class="c-bolt-logo--invert"
  style=""
>
  <bolt-image
    alt="Paypal Logo"
    ratio="124/33"
    src="/fixtures/logo-paypal.svg"
    style=""
  />
</bolt-logo>
`;

exports[`logo Logo with invert set to "true" renders properly 1`] = `
<bolt-logo class="c-bolt-logo--invert">
  <bolt-image no-lazy
              ratio="124/33"
              placeholder-color="hsl(233, 33%, 97%)"
              placeholder-image="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
              src="/fixtures/logo-paypal.svg"
              srcset="/fixtures/logo-paypal.svg"
  >
    <bolt-ratio ratio="124/33">
      <ssr-keep for="bolt-ratio"
                class="c-bolt-ratio"
                style="padding-bottom: calc((33 / 124) * 100%); --aspect-ratio: 3.7575757575758;"
      >
        <img class="c-bolt-image__image"
             alt
             srcset="/fixtures/logo-paypal.svg"
             sizes="auto"
             src="/fixtures/logo-paypal.svg"
        >
      </ssr-keep>
    </bolt-ratio>
  </bolt-image>
</bolt-logo>
`;
